src/compileModel.ts

```diff
@@ -28,13 +29,19 @@ export type ModelConstructor<TSchema extends Schema | null> = ReturnType<
 
 export type Model<TSchema extends Schema | null> = InstanceType<ModelConstructor<TSchema>>;
 
+/** Used as an intersection type to make _id & __v properties required */
+export interface RequiredModelMeta {
+	_id: string;
```

Comment:
I think `__v` could be `null` if it's a new record right? Would that mean this should be unioned with `null`? What about `_id`? Could it be `null`? Both class member variables are unioned with `null` on the dynamically created Model returned from `compileModel`.

Reply:
I see, thank you clearing that up for me! The tests help as well.